data/transactions/application_test.go

```diff
@@ -27,31 +27,38 @@ import (
 
 	"github.com/algorand/go-algorand/config"
 	"github.com/algorand/go-algorand/data/basics"
+	basics_testing "github.com/algorand/go-algorand/data/basics/testing"
 	"github.com/algorand/go-algorand/protocol"
 	"github.com/algorand/go-algorand/test/partitiontest"
 )
 
-func TestApplicationCallFieldsNotChanged(t *testing.T) {
+func TestResourceRefEmpty(t *testing.T) {
 	partitiontest.PartitionTest(t)
+	t.Parallel()
 
-	af := ApplicationCallTxnFields{}
-	s := reflect.ValueOf(&af).Elem()
-
-	if s.NumField() != 14 {
-		t.Errorf("You added or removed a field from transactions.ApplicationCallTxnFields. " +
-			"Please ensure you have updated the Empty() method and then " +
-			"fix this test")
+	assert.True(t, ResourceRef{}.Empty())
+	for _, nz := range basics_testing.NearZeros(t, ResourceRef{}) {
+		rr := nz.(ResourceRef)
+		assert.False(t, rr.Empty(), "Empty is disregarding a non-zero field in %+v", rr)
 	}
 }
 
 func TestApplicationCallFieldsEmpty(t *testing.T) {
 	partitiontest.PartitionTest(t)
+	t.Parallel()
 
-	a := require.New(t)
+	a := assert.New(t)
 
 	ac := ApplicationCallTxnFields{}
 	a.True(ac.Empty())
 
+	for _, nz := range basics_testing.NearZeros(t, ac) {
+		fields := nz.(ApplicationCallTxnFields)
+		a.False(fields.Empty(), "Empty is disregarding a non-zero field in %+v", fields)
+	}
+
+	// Everything below here ought to be redundant after the NearZeros loop.
+
```

Comment:
why not to remove then?

Reply:
I am willing.  I was leaving it there until we are comfortable with `NearZeros`.  I want at least one more person to say they're ok with it. @cce @gmalouf

Reply:
removing